src/uppaal_calls.cpp

```diff
@@ -52,19 +53,37 @@ std::string getEnvVar(std::string const &key) {
   return std::string(val);
 }
 
+std::string exec(const char *cmd) {
+  std::array<char, 128> buffer;
+  std::string result;
+  std::shared_ptr<FILE> pipe(popen(cmd, "r"), pclose);
+  if (!pipe)
+    throw std::runtime_error("popen() failed!");
+  while (!feof(pipe.get())) {
+    if (fgets(buffer.data(), 128, pipe.get()) != nullptr)
+      result += buffer.data();
+  }
+  return result;
+}
+
+
 ::std::vector<timedelta> solve(const AutomataSystem &sys, std::string file_name,
                                std::string query_str) {
   XMLPrinter printer;
   SystemVisInfo sys_vis_info(sys);
   printer.print(sys, sys_vis_info, file_name + ".xml");
   return solve(file_name, query_str);
 }
+
+
 ::std::vector<timedelta> solve(std::string file_name, std::string query_str) {
   std::vector<timedelta> res;
   std::ofstream myfile;
+  std::filesystem::remove(file_name + ".q");
```

Comment:
I think this shouldn't be necessary if you truncate the file afterwards, but of course it also doesn't hurt ;-)